### DISCUSSION TITLE
How to pass --pass-credentials for a helm type component

### DISCUSSION THREAD

**Discussion by nemo-xue** [kubevela/kubevela#3465] [Q&A] [answered]
Hi, 

For vela's helm type component, is there a way to pass helm parameter --pass-credentials (passCredentials in fluxcd)?

Issue description:
After I deployed helm component, the helmrepositories.source.toolkit.fluxcd.io shows READY. But the helmcharts.source.toolkit.fluxcd.io shows 401 error.
I need to manually add "passCredentials: true" to the helmrepositories to make 401 error go away. 

My helm repository is a private one. 
When I use helm command to add the repo, I also need to pass --pass-credentials. Otherwise, I will randomly get 401 error when I try to do helm pull.

Thanks in advance.

**Comment by nemo-xue** [comment, marked as answer]
I solved this by editing the helm component. 
Add "passCredentials: true" to HelmRepository spec.

**Reply by wonderflow** [reply]
great
